portfolio/src/main/webapp/task.js

```diff
@@ -231,8 +231,7 @@ function collectAllTasks() {
         taskDescription,
         getDurationMinutes(taskLength, taskLengthUnit),
         taskPriority);
-    const taskJson = JSON.stringify(task);
-    allTaskJson.push(taskJson);
+    allTaskJson.push(task);
```

Comment:
This is where the bug was. collectAllTasks was returning an array of JSON objects and when information is sent to the servlet it is already converted to JSON so it was unnecessary.